File: tests/fixtures/extract/sample.py
"""Fixture: methods, docstrings, lambdas, nested functions, small bodies."""


class Inventory:
    def restock(self, items):
        """Restock the inventory from a list of (name, count) pairs."""
        # merge counts into the ledger
        for name, count in items:
            current = self.ledger.get(name, 0)
            self.ledger[name] = current + count
        return len(self.ledger)


def report(inventory, minimum):
    def format_line(name, count):
        label = name.upper()
        padded = label.ljust(20)
        return f"{padded} {count}"

    lines = []
    for name, count in sorted(inventory.ledger.items()):
        if count >= minimum:
            lines.append(format_line(name, count))
    return "\n".join(lines)


def tiny():
    return 1

File: tests/fixtures/extract/Sample.cs
// Fixture: methods, constructors, local functions, lambdas, small bodies.

public class Ledger
{
    private readonly Dictionary<string, decimal> balances = new();

    public Ledger(IEnumerable<(string, decimal)> seed)
    {
        // seed the initial balances
        foreach (var (name, amount) in seed)
        {
            balances[name] = balances.GetValueOrDefault(name) + amount;
        }
    }

    public decimal Transfer(string from, string to, decimal amount)
    {
        decimal Withdraw(string account)
        {
            var available = balances.GetValueOrDefault(account);
            var taken = Math.Min(available, amount);
            balances[account] = available - taken;
            return taken;
        }

        var moved = Withdraw(from);
        balances[to] = balances.GetValueOrDefault(to) + moved;
        return moved;
    }

    public int Tiny()
    {
        return 1;
    }
}
